File: qam-ms/src/main/java/com/aura/admin/adminqamm/model/Interfaz.java
package com.aura.admin.adminqamm.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="interfaz_grafica")
public @Getter @Setter class Interfaz {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id_interfaz")
    private int idInterfaz;
    @Column(name = "nombre_interfaz")
    private String nombre;

    @Column(name="titulo_interfaz")
    private String titulo;

    @Column(name = "url_logo_icono")
    private String urlLogoIcono;
    @Column(name = "color_global_primario")
    private String colorGlobalPrimario;
    @Column(name = "color_global_secundario")
    private String colorGlobalSecundario;
    @Column(name = "color_home_primario")
    private String colorHomePrimario;
    @Column(name = "color_home_secundario")
    private String colorHomeSecundario;
    @Column(name = "color_texto")
    private String colorTexto;
    @Column(name = "color_error")
    private String colorError;
    @Column(name = "color_background")
    private String colorBackground;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="id_usuario")
    @JsonBackReference(value="user-interfaces")
    private Usuario usuario;

    @Column(name = "fecha_modificacion")
    private Date fechaModificacion;
    @Column(name = "fecha_registro")
    private Date fechaRegistro;
    @Column(name = "ind_activo")
    private boolean activo;
}

File: qam-front/src/app/model/interfaz.ts
import { DatePipe } from '@angular/common';

export class Interfaz {


    idInterfaz:number;
    nombre:string;
    titulo:string;
    urlLogoIcono:File;
    urlLogoIconoId:string;
    colorGlobalPrimario:string;
    colorGlobalSecundario:string;
    colorHomePrimario:string;
    colorHomeSecundario:string;
    colorTexto:string;
    colorError:string;
    colorBackground:string;
   
    fechaModificacion:Date;
    fechaRegistro:Date;
    activo:boolean;

    build(content:any){
        const datepipe: DatePipe = new DatePipe('en-US')
        this.idInterfaz = content.idInterfaz;
        this.nombre = content.nombre;
        this.titulo = content.titulo;
        this.urlLogoIconoId = content.urlLogoIcono;
        this.colorGlobalPrimario = content.colorGlobalPrimario;
        this.colorGlobalSecundario = content.colorGlobalSecundario;
        this.colorHomePrimario = content.colorHomePrimario;
        this.colorHomeSecundario = content.colorHomeSecundario;
        this.colorTexto = content.colorTexto;
        this.colorError = content.colorError;
        this.colorBackground = content.colorBackground;

    

  

        const fechaR:any = datepipe.transform(content.fechaRegistro, 'dd-MM-YYYY HH:mm:ss');
        this.fechaRegistro = fechaR!==null?fechaR:"";

        const fechaM:any = datepipe.transform(content.fechaModificacion, 'dd-MM-YYYY HH:mm:ss');
        this.fechaModificacion = fechaM!==null?fechaM:"";
        this.activo = content.activo;
    }
}
